src/bblocks/datacommons_tools/custom_data/data_management.py

```diff
@@ -869,26 +898,85 @@ def validate_config(self) -> CustomDataManager:
         self._config.validate_config()
         return self
 
-    def __repr__(self) -> str:
-        input_files_count = len(self._config.inputFiles)
-        sources_count = len(self._config.sources)
-        nodes_count = sum(
-            [len(var) for var in [n.nodes for n in self._mcf_nodes.values()] if var]
-        )
+    def merge_config(
+        self,
+        config: Config | dict | str | PathLike[str],
+        *,
+        policy: DuplicatePolicy = "error",
+    ) -> CustomDataManager:
+        """Merge ``config`` into the current configuration.
 
-        variables_count = len(self._config.variables or {}) + nodes_count
-        dataframes_count = len(self._data)
+        Args:
+            config: The config to merge. This can be a Config object, a dictionary,
+                or a path to a JSON file.
+            policy: How to resolve collisions. Can be "error", "override", or "ignore".
 
-        include_input_subdirs = self._config.includeInputSubdirs
-        group_statvars = self._config.groupStatVarsByProperty
+        """
 
-        return (
-            f"<CustomDataManager config: "
-            f"\n{input_files_count} inputFiles, with {dataframes_count} containing data"
-            f"\n{sources_count} sources"
-            f"\n{variables_count} variables"
-            f"\nflags: includeInputSubdirs={include_input_subdirs}, "
-            f"groupStatVarsByProperty={group_statvars}>"
+        if isinstance(config, (str, PathLike)):
+            cfg = Config.from_json(str(config))
+        elif isinstance(config, dict):
+            cfg = Config.model_validate(config)
+        else:
+            cfg = config
+
+        merge_configs(existing=self._config, new=cfg, policy=policy)
+        return self
+
+    def merge_configs_from_directory(
+        self,
+        directory: str | PathLike[str],
+        *,
+        policy: DuplicatePolicy = "error",
+        replace_loaded_config: bool = True,
+    ) -> CustomDataManager:
+        """Merge all config files found under ``directory``. This will recursively
+        search for config files in subdirectories. It will merge them with whatever
+        config is already loaded in the manager.
+
+        Args:
+            directory: The directory to search for config files.
+            policy: How to resolve collisions. Can be "error", "override", or "ignore".
```

Comment:
How do the policies work when there are multiple configs? Say you want to add 2 configs in the directory where the `includeSubdirs` is True in one and False in the other, what will the value be if policy is ignore or override?